src/go/tests/relay/in_process_relay_test.go

```diff
@@ -0,0 +1,127 @@
+package main
+
+import (
+	"bytes"
+	"context"
+	"io"
+	"net/http"
+	"sync"
+	"testing"
+	"time"
+
+	"github.com/golang/glog"
+
+	"github.com/googlecloudrobotics/core/src/go/cmd/http-relay-client/client"
+	"github.com/googlecloudrobotics/core/src/go/cmd/http-relay-server/server"
+)
+
+var (
+	port      = 8081
+	blockSize = 10 * 1024
+	once      sync.Once
+)
+
+func initRelay() {
+	once.Do(func() {
+		glog.Info("Running init.")
+
+		go func() {
+			relayServer := server.NewServer()
+			relayServer.Start(port, blockSize)
+		}()
+
+		go func() {
+			config := client.DefaultClientConfig()
+			config.BackendScheme = "http"
+			config.RelayScheme = "http"
+			config.BackendAddress = "127.0.0.1:8083"
+			config.DisableAuthForRemote = true
+			relayClient := client.NewClient(config)
+			relayClient.Start()
+		}()
+
+		relayHealthy := false
+		deadline := time.Now().Add(5 * time.Second)
+		for time.Now().Before(deadline) {
+			res, err := http.Get("http://127.0.0.1:8081/healthz")
+			if err != nil {
+				glog.Infof("Relay server is has not yet started, retrying.")
+				time.Sleep(250 * time.Millisecond)
+			} else {
+				glog.Info("Relay server is up and running.")
+				relayHealthy = true
+				defer res.Body.Close()
+				io.ReadAll(res.Body)
+				break
+			}
+		}
+		if !relayHealthy {
+			glog.Fatal("Failed to bring up http relay for unknown reason.")
+		}
+	})
+}
+
+func serveFunction(
+	f func(w http.ResponseWriter, r *http.Request)) *http.Server {
+	return serveFunctionWithTimeout(f, 10*time.Second)
+}
+
+func serveFunctionWithTimeout(
+	f func(w http.ResponseWriter, r *http.Request),
+	handlerTimeout time.Duration) *http.Server {
+	httpHandlerFunc := http.HandlerFunc(f)
+
+	srv := &http.Server{
+		Addr:         "127.0.0.1:8083",
```

Comment:
Same as above, we should be able to let the OS assign a free port and check which port is used.